### PR TITLE
Add warning on loss values when using XLA_USE_BF16

### PR DESCRIPTION
**Pull request by jysohn23**

### CODE REVIEW

File: TROUBLESHOOTING.md [hunk 1]
```diff
@@ -193,7 +193,9 @@ only be enabled for debugging.
   moving to the next step.
 
 * ```XLA_USE_BF16```: If set to 1, tranforms all the _PyTorch_ _Float_ values into _BiFloat16_
-  when sending to the _TPU_ device.
+  when sending to the _TPU_ device. Note that when using `XLA_USE_BF16=1` tensor arithmetic will
+  be done in reduced precision and so tensors such as loss values and other metrics will not be
+  useful. To get accurate loss and other metric values, user manual mixed precision.
```
Comment (taylanbil) [inline TROUBLESHOOTING.md:198]: The problem w/ loss + bf16 arises when user accumulates tensors, the numerator stops growing due to lack of precision but denom. keeps growing. so if one looks at the raw loss tensors at a given step, in bf16, that is definitely a meaningful value. but if one looks at the **average** loss over 100 steps, computed as `sum(losses) / 100`, than that's not meaningful.

Comment (jysohn23) [reply TROUBLESHOOTING.md]: Updated comments PTAL.

File: TROUBLESHOOTING.md [hunk 2]
```diff
@@ -193,7 +193,20 @@ only be enabled for debugging.
...
+  
+  ```
+  # In reduced bfloat16 precision
+  >>> torch.tensor(4096, dtype=torch.bfloat16) + torch.tensor(1, dtype=torch.bfloat16)
+  tensor(4096., dtype=torch.bfloat16)
+  # Whereas in full float32 precision
+  >>> torch.tensor(4096) + torch.tensor(1)
+  tensor(4097)
+  ```
+  So to get accurate loss and other metric values, user manual mixed precision where weights
```
Comment (taylanbil) [inline TROUBLESHOOTING.md:208]: typo: user -> use

Comment (taylanbil) [reply TROUBLESHOOTING.md:208]: weights staying in bf16 should not be that big of a problem.

File: TROUBLESHOOTING.md [hunk 3]
```diff
@@ -193,7 +193,20 @@ only be enabled for debugging.
   moving to the next step.
 
 * ```XLA_USE_BF16```: If set to 1, tranforms all the _PyTorch_ _Float_ values into _BiFloat16_
-  when sending to the _TPU_ device.
+  when sending to the _TPU_ device. Note that when using `XLA_USE_BF16=1` tensor arithmetic will
+  be done in reduced precision and so tensors such as loss values and other metrics will not be
```
Comment (taylanbil) [inline TROUBLESHOOTING.md:197]: I don't like mentioning "loss values" here. This gives the impressions that trained models aren't useful, and that's just not true.

File: TROUBLESHOOTING.md (hunk 2 above)
Comment (taylanbil) [reply TROUBLESHOOTING.md:208]: "so to get metric values such as average loss value over many steps, ..."

File: TROUBLESHOOTING.md (hunk 3 above)
Comment (jysohn23) [reply TROUBLESHOOTING.md]: done

File: TROUBLESHOOTING.md (hunk 2 above)
Comment (jysohn23) [reply TROUBLESHOOTING.md]: It does cause some convergence issues, but agreed it's not as big of deal so I'll delete that comment.
